nav2_util/include/nav2_util/simple_action_client.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_UTIL__SIMPLE_ACTION_CLIENT_HPP_
+#define NAV2_UTIL__SIMPLE_ACTION_CLIENT_HPP_
+
+#include <chrono>
+#include <stdexcept>
+#include <string>
+
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "rclcpp_action/client_goal_handle.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "action_msgs/msg/goal_status.hpp"
+
+namespace nav2_util
+{
+
+typedef enum
+{
+  SUCCEEDED,
+  FAILED,
+  RUNNING,
+  CANCELED
+} ActionStatus;
+
+template<typename ActionT>
+class SimpleActionClient
+{
+public:
+  explicit SimpleActionClient(rclcpp::Node::SharedPtr node, const std::string & action_name)
+  : node_(node)
+  {
+    action_client_ = rclcpp_action::create_client<ActionT>(node, action_name);
+  }
+
+  bool wait_for_server(std::chrono::milliseconds timeout = std::chrono::milliseconds::max())
+  {
+    return action_client_->wait_for_action_server(timeout);
+  }
+
+  ActionStatus invoke(
+    const typename ActionT::Goal & goal,
+    typename rclcpp_action::ClientGoalHandle<ActionT>::WrappedResult & result,
+    typename rclcpp_action::ClientGoalHandle<ActionT>::FeedbackCallback callback = nullptr)
+  {
+    // Send the goal
+
+    auto future_goal_handle = action_client_->async_send_goal(goal, callback);
+
+    if (rclcpp::spin_until_future_complete(node_, future_goal_handle) !=
+      rclcpp::executor::FutureReturnCode::SUCCESS)
+    {
+      throw std::runtime_error("send_goal failed");
+    }
+
+    auto goal_handle = future_goal_handle.get();
+
+    if (!goal_handle) {
+      throw std::runtime_error("Goal was rejected by the server");
+    }
+
+    // Wait for the result
+
+    auto future_result = action_client_->async_get_result(goal_handle);
+
+    if (rclcpp::spin_until_future_complete(node_, future_result) !=
```

Comment:
We should wait with an option for timeout.

Reply:
I've deleted the action_client template (see other comments; turned out to not be very useful).